### PR TITLE
Blockchain integrated

### PR DESCRIPTION
**Pull request by mujabmuneeb**

### CODE REVIEW

File: src/Header.jsx [hunk 1]
```diff
@@ -226,8 +227,26 @@ const User = ({address, setAddress, open, setOpen, toggleOpen}) => {
           setLoggingIn(true);
           try {
             const {address, profile} = await ceramicApi.login();
-            // console.log('login', {address, profile});
             setAddress(address);
+            if (address) {
+              (async () => {
+                var tokenUri = [];
+                const res = await fetch(`${apiBackend}/tokenids?address=${address}`);
```
Comment (avaer) [inline src/Header.jsx:234]: This could be one request to get the uris, rather than O(n + 1) requests. Is there a reason for doing it this way?

**Review by avaer — COMMENTED** [review]
Looks like it's headed in the right direction.

File: src/Header.jsx (hunk 1 above)
Comment (mujabmuneeb) [reply src/Header.jsx]: converted into single request. ✅

File: integration/web3/app.mjs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  try {
+    await contract.mint(await signer.getAddress(), tokenURI, {gasLimit: 200000});
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+async function drop(tokenID) {
+  const claimableVoucher = new ClaimableVoucher({contract: contract, signer: signer});
+  const timestamp = Math.round(new Date().getTime() / 1000) + 1000;
```
Comment (avaer) [inline integration/web3/app.mjs:30]: This doesn't matter, but:

```new Date().getTime() = Date.now()```

**Review by avaer — APPROVED** [review]

**Comment by avaer** [comment]
There is no description on this issue. Can you explain how it is treating the MetaMask and non-MetaMask cases?

**Comment by avaer** [comment]
I reverted this PR because it seems other PR prerequisites are not met: https://github.com/webaverse/app/commit/9180a7d52f4eeb052c866cd42645fb170bbc4181

Can you please add a description about dependencies and merge order?

**Pull request merged**